code_fellows/deque.py

```diff
@@ -0,0 +1,81 @@
+# https://codefellows.github.io/sea-python-401d6/assignments/deque.html
+
+
+class Deque:
+    def __init__(self):
+        self.head = None
+        self.tail = None
+        self._length = 0
+
+    def __len__(self):
+        return self._length
+
+    def size(self):
+        return self._length
```

Comment:
how does _length change?

Reply:
Look in the pop and append methods. https://github.com/nxnode/Data-Structures/pull/7/files#diff-7c04b501a945c4cd1ca7a748c66ed11f7f4aa7f19a06f2a5a757f3551748f8c5R32

Reply:
i guess i mean more of if you did
deque = Deque([1, 2, 3)]
does deque._length == 3?

Reply:
it would just give a TypeError because it's expecting a single argument. I get messed up with instructions, basically every single time I do these things. I either add more than it asks or miss things it requires. But the assignment said "a value" so I assumed it was not going to get an iterable passed. 

The wiki page for double ended queue doesn't really say either. Did you make yours just iter over whatever argument passed and then just implement it with the first being the last

so `Deque(["one", "two", "three"])` would have the same result as:
```
test_deq = Deque()
test.deq.appendleft("one")
test.deq.appendleft("two")
test.deq.appendleft("three")
```

or should it be append? Or does the order not even matter when it's being instantiated with an iterable?

I like the binary heap more because it's got very defined rules. Low up top and left child first (or right child if you want). No guesswork.

Reply:
The instructions didn't say anything about the starting state. So this implementation is correct.

> i guess i mean more of if you did
deque = Deque([1, 2, 3)]

Take a look at the init function.

> it would just give a TypeError because it's expecting a single argument.

yup, and the one argument it expects is `self`..
`TypeError: Deque.__init__() takes 1 positional argument but 2 were given`

Reply:
> or should it be append? Or does the order not even matter when it's being instantiated with an iterable?

Generally, when you are creating something from a list you do it from start to end. So index one from the list would go onto the deque first.  Even though it is double ended it is still a queue and pop should remove items in the order the were put on the queue.  popleft removes them in reverse order( like pulling from the end of the line instead of the front).  So when initializing with values `append` would be used.